dbms/src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -120,7 +121,8 @@ MergeTreeData::MergeTreeData(
     data_parts_by_info(data_parts_indexes.get<TagByInfo>()),
     data_parts_by_state_and_info(data_parts_indexes.get<TagByStateAndInfo>())
 {
-    setPrimaryKeyIndicesAndColumns(order_by_ast_, primary_key_ast_, columns_, indices_);
+    setPrimaryKeyIndicesAndColumns(order_by_ast_, primary_key_ast_, columns_, indices_, constraints_);
+    setConstraints(constraints_);
```

Comment:
We will call it in `setPrimaryKeyIndicesAndColumns`. Also need to rename `setPrimaryKeyIndicesAndColumns` -> `setProperties`?.